### PR TITLE
[Store] Add self-sign up feature in UI

### PR DESCRIPTION
**Pull request by vithu30**
This PR contains implementation to add self-sign up feature via UI. Currently, in APIM 3.0.0 we do not have way to create new user via UI.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationsAPI.java [hunk 1]
```diff
@@ -76,4 +77,12 @@ public Response availableFeatures() {
...
         return Response.ok(featureMap, MediaType.APPLICATION_JSON).build();
     }
+
+    @GET
+    @Path("/policyRoutes")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response policyRoutes() {
+        PolicyRoutes policyRoutes = ConfigurationService.getInstance().getApimUIConfigurations().getPolicyRoutes();
+        return Response.ok(policyRoutes, MediaType.APPLICATION_JSON).build();
+    }
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationsAPI.java]: Shall we add java docs here?

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.configurations/src/main/java/org/wso2/carbon/apimgt/rest/api/configurations/ConfigurationsAPI.java]: Added

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java [hunk 2]
```diff
@@ -161,22 +162,29 @@ public JsonObject getAuthenticationConfigurations(String appName)
...
         log.debug("Received consumer key & secret for {} application.", appName);
         try {
-            if (KeyManagerConstants.AUTHORIZATION_CODE_GRANT_TYPE.equals(grantType)) {
+            if (KeyManagerConstants.PASSWORD_GRANT_TYPE.equals(grantType) ||
+                    KeyManagerConstants.REFRESH_GRANT_TYPE.equals(grantType) ||
+                    KeyManagerConstants.CLIENT_CREDENTIALS_GRANT_TYPE.equals(grantType)) {
+                accessTokenRequest = AuthUtil
+                        .createAccessTokenRequest(userName, password, grantType, refreshToken,
+                                null, validityPeriod, scopesList,
+                                consumerKeySecretMap.get("CONSUMER_KEY"), consumerKeySecretMap.get("CONSUMER_SECRET"));
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java]: Shall we use Constants? :)

Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java]: Changed accordingly

**Comment by Mariangela** [comment]
@npamudika & @vithu30 , please mark if this has a doc impact or not by selecting the appropriate label (Docs/Has Impact OR Docs/No Impact).

**Pull request merged**